src/lib.rs

```diff
@@ -151,11 +140,10 @@ pub fn init_tracing_with_fallbacks(
         propagation::TextMapSplitPropagator::from_env().expect("setup of Propagation"),
     );
 
-    let otel_layer = tracing_opentelemetry::layer()
-        .with_tracer(otel_tracer)
-        .with_filter(filter::OtelFilter::default());
+    let otel_layer = tracing_opentelemetry::layer().with_tracer(otel_tracer);
     let subscriber = tracing_subscriber::registry()
-        .with(build_logger_text(log_level))
+        .with(Into::<filter::TracingFilter>::into(log_level))
```

Comment:
This line is the actual fix.

But I also had to rewrite our custom `TracingFilter` struct to be able to use it here like that.

Reply:
Previously the filtering was done in the `otel_layer` layer, but it's not correct